js/models/conversations.js

```diff
@@ -415,10 +416,17 @@
           text: this.lastMessage,
         },
         isOnline: this.isOnline(),
+        isMe: this.isMe(),
+        hasNickname: !!this.getNickname(),
 
         onClick: () => this.trigger('select', this),
         onBlockContact: () => this.block(),
         onUnblockContact: () => this.unblock(),
+        onChangeNickname: () => this.changeNickname(),
+        onClearNickname: async () => this.setNickname(null),
```

Comment:
Was copied from
```
onClearNickname: async () => {
  this.model.setNickname(null);
},
```
Will this change the behaviour at all?

Reply:
Not sure, but this looks like a mistake in conversation_view.js.
It would make sense to let the caller choose to wait or not for a promise.
So either you remove the brackets `() => this.setNickname(null)` which implicitly returns what `setNickname` returns or you explicitly return it:
```js
() => {
  return this.model.setNickname(null);
}
```
(I'm guessing that `setNickname` returns a promise)
This way you don't need `async` unless you actually call `await` in that function.

Reply:
implicitly returning it is fine `() => this.setNickname(null)`.
As sacha said, you don't really need an async unless:
1. You need synchronous code to be async (which would be the same as wrapping it in a promise. Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)
2. You need to `await` another promise in the block
3. You need to override the return value of the promise in the block (E.g if you want to return nothing regardless of what `setNickname` returns)

If the function you call is going to return a promise anyway then you can just call it and let it return its promise.

Reply:
Kk should I update all the other props to follow this pattern while I'm at it?

Reply:
ye if needed :)